src/viewport.cpp

```diff
@@ -224,6 +224,57 @@ class VoxelCollector {
  * @ingroup viewport_group
  */
 struct DrawData {
+	/**
+	 * Default constructor initializing only the fields that mostly always have the same value.
+	 */
+	DrawData()
+	{
+		this->recolour = nullptr;
+		this->highlight = false;
+	}
+
+	DrawData(const DrawData &dd)
+	{
+		this->level = dd.level;
+		this->z_height = dd.z_height;
+		this->order = dd.order;
+		this->sprite = dd.sprite;
+		this->base = dd.base;
+		this->recolour = dd.recolour;
+		this->highlight = dd.highlight;
+	}
+
+	DrawData &operator=(const DrawData &dd)
+	{
+		if (this != &dd) {
+			this->level = dd.level;
+			this->z_height = dd.z_height;
+			this->order = dd.order;
+			this->sprite = dd.sprite;
+			this->base = dd.base;
+			this->recolour = dd.recolour;
+			this->highlight = dd.highlight;
+		}
+		return *this;
+	}
```

Comment:
Also, a bit out of scope of patch, but maybe z_height & base could be combined - we do have an XYZPoint type, after all